src/gui/sdlmain.cpp

```diff
@@ -1949,20 +1944,21 @@ void Config_Add_SDL() {
 }
 
 static void show_warning(char const * const message) {
-	bool textonly = true;
-#ifdef WIN32
-	textonly = false;
-	if ( !sdl.inited && SDL_Init(SDL_INIT_VIDEO|SDL_INIT_NOPARACHUTE) < 0 ) textonly = true;
-	sdl.inited = true;
+#ifndef WIN32
+	printf("%s", message);
+	return;
 #endif
-	printf("%s",message);
-	if(textonly) return;
-	if (!sdl.window)
-		if (!GFX_SetSDLSurfaceWindow(640, 400))
-			return;
-	sdl.surface = SDL_GetWindowSurface(sdl.window);
-	if(!sdl.surface)
+	if ( !sdl.inited && SDL_Init(SDL_INIT_VIDEO|SDL_INIT_NOPARACHUTE) < 0 ) {
+		sdl.inited = true;
+		printf("%s",message);
 		return;
+	} else {
+		if (!sdl.window && !GFX_SetSDLSurfaceWindow(640, 400))
+			return;
+		sdl.surface = SDL_GetWindowSurface(sdl.window);
+		if (!sdl.surface)
+			return;
+	}
```

Comment:
There's a return in `if` part of the statement, so we don't need else any more :)